src/app/service/scoap-squad/ScoapDatabase.ts

```diff
@@ -0,0 +1,106 @@
+import dbInstance from '../../utils/db';
+import { Db } from 'mongodb';
+import cloneDeep from 'lodash.clonedeep';
+import { ScoapEmbed, VoteRecord } from './ScoapClasses';
+import client from '../../app';
+import { TextChannel } from 'discord.js';
+import channelIds from '../constants/channelIds';
+import ScoapUtils from '../../utils/ScoapUtils';
+import constants from '../constants/constants';
+import { createReactionCollector, collectReactions } from './ScoapPoll';
+
+// ScoapSquad state initialization
+export const scoapEmbedState = {};
+export const botConvoState = {};
+export const voteRecordState = {};
+ScoapUtils.logToFile(`state objects initiated. \n scoapEmbedState: ${JSON.stringify(scoapEmbedState)} \n botConvoState: ${JSON.stringify(botConvoState)}  \n voteRecordState: ${JSON.stringify(voteRecordState)}`);
+setInterval(function() { ScoapUtils.purgeExpiredBotConvo(botConvoState); }, 60000);
+
+
+const decircularizeScoapEmbed = (scoapEmbed) => {
+	const clone = cloneDeep(scoapEmbed);
+	clone.setCurrentMessage(scoapEmbed.getCurrentMessage().id);
+	clone.setCurrentChannel(scoapEmbed.getCurrentChannel().id);
+	return clone;
+};
+
+export const updateScoapEmbedAndVoteRecordDb = async (scoapEmbed, voteRecord) => {
+	const clone = decircularizeScoapEmbed(scoapEmbed);
+	const filter = { id: scoapEmbed.getId() };
+	const options = { upsert: true };
+	const updateDoc = { $set: { scoapEmbed: clone, voteRecord: voteRecord } };
+	const db: Db = await dbInstance.dbConnect('degen');
```

Comment:
Can we use constants file instead of 'degen'?

Reply:
Not sure what you mean, can you elaborate a bit more? Is constants another database? I put it in the location recommended by @nonsensecodes.

Reply:
oh i see what you mean now, if it is just about the string variable, yes done